Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1121,7 +1121,20 @@ class BrowserViewController: UIViewController {
     }
 
     override func observeValue(forKeyPath keyPath: String?, of object: Any?, change: [NSKeyValueChangeKey: Any]?, context: UnsafeMutableRawPointer?) {
-        guard let webView = object as? WKWebView, let tab = tabManager[webView], let kp = keyPath, let path = KVOConstants(rawValue: kp) else {
+        
+        guard let webView = object as? WKWebView else {
+            log.error("An object of type: \(String(describing: object)) is being observed instead of a WKWebView")
+            return  //False alarm.. the source MUST be a web view.
+        }
+        
+        //WebView is a zombie and somehow still has an observer attached to it
+        guard let tab = tabManager[webView] else {
+            log.error("WebView: \(webView) has been removed from TabManager but still has attached observers")
+            return
+        }
+        
+        //Must handle ALL keypaths
```

Comment:
nitpick: please stick to one style for comments, sometimes you add a space after `//` sometimes not :D
we usually add spaces